weld/passes.rs

```diff
@@ -0,0 +1,67 @@
+use super::ast::*;
+use super::error::*;
+use super::transforms;
+
+
+pub struct Pass {
+    transform_fns: Vec<fn(&mut Expr<Type>)>,
+    transform_names: Vec<String>,
```

Comment:
This is kind of hard to maintain because the vectors have to be the same length; I'd create a separate "Transformation" type that has both a name and a function, or drop the names.